Fix a file descriptor leak:

https://github.com/xinetd-org/xinetd/issues/23

Patch copied from Debian:

https://anonscm.debian.org/cgit/collab-maint/xinetd.git/tree/debian/patches/000012-fix_fd_leak

Patch sent upstream at https://github.com/xinetd-org/xinetd/pull/26.

diff --git a/xinetd/xgetloadavg.c b/xinetd/xgetloadavg.c
index 5a26214..fe0f872 100644
--- a/xinetd/xgetloadavg.c
+++ b/xinetd/xgetloadavg.c
@@ -34,7 +34,7 @@ double xgetloadavg(void)
 
    if( fscanf(fd, "%lf", &ret) != 1 ) {
       perror("fscanf");
-      return -1;
+      ret = -1;
    }
 
    fclose(fd);
-- 
2.7.4

